src/main/java/io/aiven/kafka/connect/common/output/OutputWriter.java

```diff
@@ -19,28 +19,49 @@
 import java.io.IOException;
 import java.io.OutputStream;
 import java.util.Collection;
+import java.util.Map;
 import java.util.Objects;
+import java.util.zip.GZIPOutputStream;
 
+import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.sink.SinkRecord;
 
+import io.aiven.kafka.connect.common.config.CompressionType;
+import io.aiven.kafka.connect.common.config.FormatType;
 import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.output.jsonwriter.JsonLinesOutputWriter;
+import io.aiven.kafka.connect.common.output.jsonwriter.JsonOutputWriter;
+import io.aiven.kafka.connect.common.output.plainwriter.PlainOutputWriter;
+
+import com.github.luben.zstd.ZstdOutputStream;
+import org.xerial.snappy.SnappyOutputStream;
 
 public abstract class OutputWriter implements AutoCloseable {
 
-    private OutputStream outputStream;
+    private final OutputStreamWriter writer;
+
+    protected final OutputStream outputStream;
+
     private Boolean isOutputEmpty;
+
     private Boolean isClosed;
-    private OutputStreamWriter writer;
 
-    public OutputWriter(final Collection<OutputField> fields,
-                        final OutputStream outputStream) {
-        this.writer = writer(fields);
+    protected OutputWriter(final OutputStream outputStream, final OutputStreamWriter writer) {
+        Objects.requireNonNull(writer, "writer");
+        Objects.requireNonNull(outputStream, "outputStream");
+        this.writer = writer;
         this.outputStream = outputStream;
         this.isOutputEmpty = true;
         this.isClosed = false;
     }
 
-    public void writeRecord(final SinkRecord record) throws IOException {
```

Comment:
Please note that the downstream connectors rely on this method to be public. This will most likely break them, unless there's some plan for this.

Reply:
It will break them in any case since OutputWriter now builds from builder, but I will change GCS and S3 when I will add parquet